scriptmodules/supplementary/golang.sh

```diff
@@ -0,0 +1,45 @@
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+rp_module_id="golang"
+rp_module_desc="Golang binary install"
+rp_module_section="opt"
+
+function depends_golang() {
+    getDepends curl tar
+}
+
+function remove_golang() {
+    if [[ -d "${md_inst}/go" ]]; then
+        rm -rf ${md_inst}/go
+    fi
+}
+
+function install_bin_golang() {
+    local version=""
+    if [[ -e "${md_inst}/go/bin/go" ]]; then
+        local version=$(GOROOT=${md_inst}/go ${md_inst}/go/bin/go version | sed -e 's/.*\(go1[^ ]*\).*/\1/')
+    fi
+    echo Current Go version: $version
+    if [[ ! "${version}" < "go1.8" ]]; then
+        return 0
+    fi
+    remove_golang
+    local arch="armv6l"
+    if isPlatform "x86"; then
+        if [[ $(getconf LONG_BIT) == "64" ]]; then
+            arch="amd64"
+        else
+            arch="386"
+        fi
+    fi
+    echo Downloading go1.8.linux-$arch.tar.gz
+    curl -o ${md_inst}/golang.tar.gz https://storage.googleapis.com/golang/go1.8.linux-$arch.tar.gz
+    tar -C ${md_inst} -xzf ${md_inst}/golang.tar.gz
+    rm ${md_inst}/golang.tar.gz
```

Comment:
Please quote these and use "$md_inst" rather than "${md_inst}" (we use the latter only when really needed - perhaps the second form is better, but it's not consistent with the rest of the code).